rskj-core/src/main/java/co/rsk/peg/federation/Federation.java

```diff
@@ -50,26 +51,27 @@ public abstract class Federation {
     protected Address address;
 
     protected Federation(
-        List<FederationMember> members,
-        Instant creationTime,
-        long creationBlockNumber,
-        NetworkParameters btcParams,
+        FederationArgs federationArgs,
         int formatVersion
     ) {
         // Sorting members ensures same order of federation members for same members
         // Immutability provides protection against unwanted modification, thus making the Federation instance
         // effectively immutable
-        this.members = Collections.unmodifiableList(members.stream().sorted(FederationMember.BTC_RSK_MST_PUBKEYS_COMPARATOR).collect(Collectors.toList()));
-        this.creationTime = creationTime.truncatedTo(ChronoUnit.MILLIS);
-        this.creationBlockNumber = creationBlockNumber;
-        this.btcParams = btcParams;
+        this.federationArgs = federationArgs;
+        this.members = Collections.unmodifiableList(federationArgs.members.stream().sorted(FederationMember.BTC_RSK_MST_PUBKEYS_COMPARATOR).collect(Collectors.toList()));
+        this.creationTime = federationArgs.creationTime.truncatedTo(ChronoUnit.MILLIS);
+        this.creationBlockNumber = federationArgs.creationBlockNumber;
+        this.btcParams = federationArgs.btcParams;
```

Comment:
I wouldn't have a `FederationArgs` object and also other variables whose values are obtained from the FederationArgs. That's duplicating the information and can lead to errors and inconsistencies.

I would simply rebuild the FederationArgs object inside `getArgs` methods

Reply:
I agree on not having a `FederationArgs` object as part of the `Federation` class and also on creating a `getArgs` method inside that build a `FederationArgs`  object.